src/commands/config.ts

```diff
@@ -1,27 +1,30 @@
-import { MessageEmbed } from "discord.js";
+import { GuildMember, MessageEmbed } from "discord.js";
 import { writeFile } from "fs";
 import type { Command } from "../command";
-import { ephemeral } from "../ephemeral";
-import { defaultArgumentParser } from "../parsers";
 
 const command: Command = {
 	name: "config",
 	description: "Gets and updates config values, do not specify a name to list all config entries.",
 	options: [
 		{
+			type: "STRING",
 			name: "name",
 			description: "The config name (case-sensitive).",
-			optional: true,
+			required: true,
 		},
 		{
+			type: "STRING",
 			name: "value",
 			description: "The new value for the config.",
-			optional: true,
+			required: true,
 		},
 	],
-	hasPermission: (state, message) => !!message.member?.roles.cache.has(state.config.developerRoleId),
-	parseArguments: defaultArgumentParser,
-	handler: (state, message, name, value) => {
+	hasPermission: (state, interaction) =>
+		(interaction.member as GuildMember).roles.cache.has(state.config.developerRoleId),
```

Comment:
I was hoping that `!!interaction.member?.roles.cache.has(state.config.developerRoleId)` would work but for some reason TS resolves the type of member as `any` instead of how discord.js defines it (as `public member: GuildMember | RawInteractionGuildMember | null`). Oh well, this is fine as it is and hopefully can be improved in the future when discord.js updates their typedefs.